cypress/e2e/e2e.cy.js

```diff
@@ -0,0 +1,140 @@
+describe('Teste E2E EngageSphere', () => {
+    beforeEach(() => {
+        cy.visit('http://localhost:3000')
+        cy.contains('button', 'Accept').click();
+    })
+
+    it('Mantém os filtros ao voltar da visualização de detalhes do cliente', () => {
+        cy.get('select[data-testid="size-filter"]').select('Very Large Enterprise')
+        cy.get('button[aria-label="View company: Lowe Co"]').click()
+        cy.get('button').contains('Back').should('be.visible')
+        cy.contains('button', 'Back').click();
+        cy.contains('Below is our customer list.').should('be.visible')
+        cy.get('select[data-testid="size-filter"]').should('have.value', 'Very Large Enterprise');
+    })
+
+    it('Retorna à lista de clientes ao clicar no botão "Voltar"', () => {
+        cy.get('button[aria-label="View company: Lowe Co"]').click();
+        cy.get('button').contains('Back').should('be.visible')
```

Comment:
Aqui podia ser algo mais simples.

Assim:

```suggestion
        cy.contains('button', 'Back').should('be.visible')
```

Perceba que há um encadeamento de comandos a menos, e a linha é mais curta.

A propósito, recomendo a leitura do seguinte conteúdo: [Como identificar um elemento por seu texto com Cypress](https://talkingabouttesting.com/2021/02/13/como-identificar-um-elemento-por-seu-texto-com-cypress/).